data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/CidrExpressionFunctionTest.java

```diff
@@ -0,0 +1,148 @@
+
+    @Test
+    void testIpv4NonMatchWithMultipleCidrBlocks() {
+        String network1 = "\"192.0.2.0/24\"";
+        String network2 = "\"192.168.1.0/24\"";
+        String network3 = "\"10.0.1.0/24\"";
+        testEvent = createTestEvent(Map.of("sourceIp", "192.0.5.3"));
+        Object expressionResult = cidrExpressionFunction.evaluate(
+                List.of("/sourceIp", network1, network2, network3), testEvent, testFunction);
+        assertThat((boolean)expressionResult, equalTo(false));
+    }
+
+    @ParameterizedTest
+    @MethodSource("ipv6AddressesInRange")
+    void testIpv6MatchWithSingleCidrBlock(String testIp) {
+        String network = "\"2001:0db8::/32\"";
+        testEvent = createTestEvent(Map.of("sourceIp", testIp));
+        Object expressionResult = cidrExpressionFunction.evaluate(List.of("/sourceIp", network), testEvent, testFunction);
+        assertThat((boolean)expressionResult, equalTo(true));
+    }
+
+    @ParameterizedTest
+    @MethodSource("ipv6AddressesInRange")
+    void testIpv6MatchWithMultipleCidrBlocks(String testIp) {
+        String network1 = "\"2001:0db8::/32\"";
+        String network2 = "\"2001:aaaa::/32\"";
+        String network3 = "\"2001:bbbb:cccc::/48\"";
+        testEvent = createTestEvent(Map.of("sourceIp", testIp));
+        Object expressionResult = cidrExpressionFunction.evaluate(List.of("/sourceIp", network1, network2, network3), testEvent, testFunction);
+        assertThat((boolean)expressionResult, equalTo(true));
+    }
+
+    @Test
+    void testIpv6NonMatchWithMultipleCidrBlocks() {
+        String network1 = "\"2001:0db8::/32\"";
+        String network2 = "\"2001:aaaa::/32\"";
+        String network3 = "\"2001:bbbb:cccc::/48\"";
+        testEvent = createTestEvent(Map.of("sourceIp", "2001:dddd:aaaa:bbbb::"));
+        Object expressionResult = cidrExpressionFunction.evaluate(List.of("/sourceIp", network1, network2, network3), testEvent, testFunction);
+        assertThat((boolean)expressionResult, equalTo(false));
+    }
+
+    @Test
+    void testTooFewArgumentsThrowsException() {
+        testEvent = createTestEvent(Map.of("sourceIp", "192.0.2.3"));
+        assertThrows(IllegalArgumentException.class,
+                () -> cidrExpressionFunction.evaluate(List.of("/sourceIp"), testEvent, testFunction));
+    }
+
+    @Test
+    void testArgumentTypeNotSupportedThrowsException() {
+        testEvent = createTestEvent(Map.of("sourceIp", "192.0.2.3"));
+        assertThrows(IllegalArgumentException.class,
+                () -> cidrExpressionFunction.evaluate(List.of("/sourceIp", 123), testEvent, testFunction));
+    }
+
+    @Test
+    void testIpAddressNotExistInEventThrowsException() {
+        String network = "\"192.0.2.0/24\"";
+        testEvent = createTestEvent(Map.of("sourceIp", "192.0.2.3"));
+        assertThrows(NullPointerException.class,
+                () -> cidrExpressionFunction.evaluate(List.of("/destinationIp", network), testEvent, testFunction));
```

Comment:
Should this throw?

Reply:
I don't think it should throw if it doesn't exist. It should return false, I think.